### ISSUE TITLE
Fix `url` and `thread_url` in Combined Summary Documents

### ISSUE THREAD

**Issue by kouloumos** [bitcoinsearch/summarizer#64]
The "Push Combined Summary From XML Files to ES INDEX" cron job is currently pushing combined summary documents to the Elasticsearch index with incorrect URLs. Specifically, the `url` and `thread_url` fields in each summary document are being set to the link of the last reply in the resource, rather than the correct, original link.

### Background

The URLs in question are generated in the [`read_xml_file` method](https://github.com/bitcoinsearch/summarizer/blob/main/src/xml_utils.py#L71) as part of the XML processing workflow. This issue originates from the "XML Generation" cron job, which generates these XML files with the incorrect `url` [here](https://github.com/bitcoinsearch/summarizer/blob/main/src/xml_utils.py#L446).

### Proposed Solution

Review the URL generation logic in the XML generation cron job to ensure that `link` points to the correct resource link rather than the last reply in the resource. The link that we want here is actually the `thread_url`.

**Comment by urvishp80** [comment]
@kouloumos based on your Proposed Solution for the [script here](https://github.com/bitcoinsearch/summarizer/blob/ad63d798dcee6fdf51f1016859e26ec14e8abcf9/src/xml_utils.py#L446) I'm thinking of updating the value of `url` key by the link of main/first post for the given title.
```
feed_data = {
          ...
          'url': <The link of first/main post (instead of latest post)>,
          ...
          } 
```

**Comment by kouloumos** [comment]
> I'm thinking of updating the value of `url` key by the link of main/first post for the given title.
> 
> ```
> feed_data = {
>           ...
>           'url': <The link of first/main post (instead of latest post)>,
>           ...
>           } 
> ```

Will that result in the `thread_url` of the combined summary to match the `thread_url` of the individual documents of the thread? If yes, then proceed wit that.

**Comment by kouloumos** [comment]
fixed by #68 

@urvishp80 fyi: Here are [some keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) that you can use when creating a PR to link it with an existing issue. Therefore when the PR is merged, the issue is also closed automatically.
